service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -915,4 +922,24 @@ public void approveSiteImage(JWTData userData, int imageID) {
     imageRecord.setApprovalStatus(ImageApprovalStatus.APPROVED.getApprovalStatus());
     imageRecord.store();
   }
+
+  @Override
+  public void rejectSiteImage(JWTData userData, int imageId, RejectImageRequest rejectImageRequest) {
+    checkImageExists(imageId);
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+
+    if (rejectImageRequest.getRejectionReason() != null) {
+      int uploaderId  = db.select(SITE_IMAGES.UPLOADER_ID)
+                      .from(SITE_IMAGES)
+                      .where(SITE_IMAGES.ID.eq(imageId))
+                      .fetchOne(0, int.class);
+
+      UsersRecord user = db.selectFrom(USERS).where(USERS.ID.eq(uploaderId)).fetchOne();
+      String userEmail = user.getEmail();
+      String userFullName =
+              AuthDatabaseOperations.getFullName(user.into(Users.class));
+      emailer.sendRejectImageEmail(userEmail, userFullName, rejectImageRequest.getRejectionReason());
+    }
```

Comment:
⛏️ I think we should send a rejection email regardless of if a reason was added just so to keep users updated - we could probably just add a default rejection message. Also, can we check if the reason is an empty string as well so we don't accidentally send no message?
Sorry if the API spec wasn't clear about this - that's totally my bad